src/billing.ts

```diff
@@ -0,0 +1,166 @@
+import { Connection } from "./connection";
+
+export interface AccountDetails {
+    firstname: string,
+    lastname: string,
+    email: string,
+    phone: string,
+    line1: string,
+    line2: string,
+    city: string,
+    state: string,
+    country: string,
+    zipcode: string
+}
+
+
+export class Billing {
+    protected _connection: Connection;
+    protected tenantName: string;
+
+    constructor(connection: Connection, tenantName: string) {
+        this._connection = connection;
+        this.tenantName = tenantName;
+    }
+
+    getAccountDetails() {
+        return this._connection.request(
+            {
+                method: "GET",
+                path: "/_api/billing/account",
+                headers: { tenant: this.tenantName },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    updateAccountDetails(accountDetails: AccountDetails) {
+        return this._connection.request(
+            {
+                method: "PUT",
+                path: "/_api/billing/contact",
+                headers: { tenant: this.tenantName },
+                body: accountDetails,
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    updatePaymentSettings(paymentMethodId: string) {
+        return this._connection.request(
+            {
+                method: "PUT",
+                path: "/_api/billing/paymentsettings",
+                headers: { tenant: this.tenantName },
+                body: { payment_method_id: paymentMethodId },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getPaymentDetailsOfPreviousMonths(limit: number) {
```

Comment:
why it is not optional.

Reply:
limit: Number of previous months for which payment details are required